src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -781,6 +781,23 @@ protected theorem extractLsb_ofNat (x n : Nat) (hi lo : Nat) :
     (extractLsb' start len x).getLsbD i = (i < len && x.getLsbD (start+i)) := by
   simp [getLsbD, Nat.lt_succ]
 
+@[simp] theorem getMsbD_extractLsb' {start len : Nat} {x : BitVec w} {i : Nat} :
+    (extractLsb' start len x).getMsbD i
+    = (decide (i < len) &&
+    (decide (start + (len - 1 - i) < w) && x.getMsbD (w - 1 - (start + (len - 1 - i))))) := by
```

Comment:
I wish I could understand what the condition was doing; While I believe the condition, maybe a comment explaining how this `getMsbD` was arrived at (ascii art?) would be super helpful.

Reply:
I agree this is the right condition. However it is not formatted correctly. This is better:
```
    (extractLsb' start len x).getMsbD i = 
      (decide (i < len) &&
      (decide (start + (len - 1 - i) < w) && 
      x.getMsbD (w - 1 - (start + (len - 1 - i))))) := by
```